src/main/java/gtclassic/common/util/GTIFilters.java

```diff
@@ -123,8 +123,15 @@ public boolean matches(ItemStack stack) {
 			if (stack.isEmpty() || this.tile.getCurrentFilter() == null) {
 				return false;
 			}
-			return this.tile.invertFilter ? !GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter())
-					: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter());
+			if (this.tile.getCurrentFilter().equals("dust")){
+				return this.tile.invertFilter ? !GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) || GTHelperStack.oreDictStartsWith(stack, "dustTiny") || GTHelperStack.oreDictStartsWith(stack, "dustSmall")
+						: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) && !GTHelperStack.oreDictStartsWith(stack, "dustTiny") && !GTHelperStack.oreDictStartsWith(stack, "dustSmall");
+			}
+			if (this.tile.getCurrentFilter().equals("ingot")){
+				return this.tile.invertFilter ? !GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) || GTHelperStack.oreDictStartsWith(stack, "ingotHot")
+						: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter()) && !GTHelperStack.oreDictStartsWith(stack, "ingotHot");
+			}
+			return this.tile.invertFilter != GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter());
```

Comment:
what is this last line? this makes no sense

Reply:
oh that was a sonar lint suggestion. if you think about it, it is basically the same as 
```
this.tile.invertFilter ? !GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter())
					: GTHelperStack.oreDictStartsWith(stack, this.tile.getCurrentFilter());``` 
lets use dust for an example
if invert filter is false, and the stack is not dust(false), it returns false.
if invert filter is false and the stack is dust(true), it returns true.
if invert filter is true, and the stack is not dust(false), it returns true.
if invert filter is true, and the stack is dust(true), it returns false
so therefore that logic is actually the same and cleaner